src/server_manager/web_app/digitalocean_app/create_server_app.ts

```diff
@@ -0,0 +1,137 @@
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+import {EventEmitter} from 'eventemitter3';
+import {customElement, html, LitElement, property} from 'lit-element';
+
+import {makePublicEvent} from '../../infrastructure/dom_events';
+import {ManagedServer, ManagedServerRepository, RegionId} from '../../model/server';
+import * as digitalocean_server from '../digitalocean_server';
+import {OutlineNotificationManager} from '../ui_components/outline-notification-manager';
+import {Location, OutlineRegionPicker} from '../ui_components/outline-region-picker-step';
+import {XhrError} from "../../cloud/digitalocean_api";
+
+// DigitalOcean mapping of regions to flags
+const FLAG_IMAGE_DIR = 'images/flags';
+const DIGITALOCEAN_FLAG_MAPPING: {[cityId: string]: string} = {
+  ams: `${FLAG_IMAGE_DIR}/netherlands.png`,
+  sgp: `${FLAG_IMAGE_DIR}/singapore.png`,
+  blr: `${FLAG_IMAGE_DIR}/india.png`,
+  fra: `${FLAG_IMAGE_DIR}/germany.png`,
+  lon: `${FLAG_IMAGE_DIR}/uk.png`,
+  sfo: `${FLAG_IMAGE_DIR}/us.png`,
+  tor: `${FLAG_IMAGE_DIR}/canada.png`,
+  nyc: `${FLAG_IMAGE_DIR}/us.png`,
+};
+
+@customElement('digitalocean-create-server-app')
+export class DigitalOceanCreateServerApp extends LitElement {
+  // External events
+  public static EVENT_SERVER_CREATE_STARTED = 'DigitalOceanCreateServerApp#ServerCreateStarted';
+  public static EVENT_SERVER_CREATE_FAILED = 'DigitalOceanCreateServerApp#ServerCreateFailed';
+  public static EVENT_AUTHORIZATION_ERROR = 'DigitalOceanCreateServerApp#AuthorizationError';
+  // Internal events
+  private static EVENT_CREATE_SERVER_REQUESTED = 'DigitalOceanCreateServerApp#_RequestCreateServer';
+
+  @property({type: Function}) localize: Function;
+
+  private notificationManager: OutlineNotificationManager;
+  private eventEmitter = new EventEmitter();
+
+  constructor() {
+    super();
+    this.addEventListener(OutlineRegionPicker.EVENT_REGION_SELECTED, (event: CustomEvent) => {
+      this.eventEmitter.emit(
+          DigitalOceanCreateServerApp.EVENT_CREATE_SERVER_REQUESTED, event.detail.regionId);
+    });
+  }
+
+  render() {
+    return html`<outline-region-picker-step .localize=${
+        this.localize}></outline-region-picker-step>`;
+  }
+
+  setNotificationManager(notificationManager: OutlineNotificationManager) {
+    this.notificationManager = notificationManager;
+  }
+
+  async start(digitalOceanServerRepository: ManagedServerRepository): Promise<void> {
+    await this.showRegionPicker(digitalOceanServerRepository);
+
+    this.eventEmitter.once(
```

Comment:
This is not needed. All you need is to have a handler for EVENT_REGION_SELECTED that is set one in the render call.

I also don't understand why you need sever created and create failure events. Who handles them? This app is supposed to be self-contained.

The only event the parent cares about is a "ServerCreated(Server)" event, which may be emitted by this app, or by the repository. We can make the app emit it for now to keep things a bit simpler.